### PR TITLE
Support year-month versions instead of just year-month-day

### PR DESCRIPTION
**Pull request by heaths**

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]

**Comment by heaths** [comment]
To note, I was successfully able to run this against a previously-problematic service using a year-month date for the new version as well. The old version was a year-month-day, but there was a year-month date in the `readme.md` that appeared to be causing problems.

**Review by mikeharder — APPROVED** [review]

File: eng/scripts/Copy-ApiVersion.ps1 [hunk 1]
```diff
@@ -38,18 +38,25 @@ param (
     [Parameter(Mandatory = $true)]
     [ArgumentCompleter({ "preview/", "stable/" })]
     [ValidateScript({ 
-            function script:Get-Version([string] $version) {
-                if ($version -match '^(?<status>stable|preview)/(?<version>(\d{4}-\d{2}-\d{2}|\d+\.\d+)(-preview(\.\d+)?)?)$') {
-                    return $Matches['status'], $Matches['version']
-                }
-        
-                throw 'Version must start with "stable/" or "preview/" and end with either a date-based version (recommended) like 2024-01-05 or a major.minor semver, followed by an optional "-preview" for previews APIs.'
+        function script:Get-Version($version) {
```
Comment (benbp) [inline eng/scripts/Copy-ApiVersion.ps1:41]: Can you get rid of the function declaration here too given you're inlining the same logic in the second validation script?

Comment (heaths) [reply eng/scripts/Copy-ApiVersion.ps1:41]: It's used in the body as well. This used to work having just once here, but PowerShell seems to have broken that. I don't want to define it thrice, and validation scripts can't use functions defined in the body or not yet imported.

**Pull request merged**